# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../grub/no-menu-flicker.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- grub-0.97/stage2/stage2.c.vanilla	2011-01-13 18:01:24.012337848 +0000
+++ grub-0.97/stage2/stage2.c 2011-01-13 18:01:24.012337848 +0000
@@ -544,5 +544,12 @@
         }
 
+      /* If we count-down a timeout, poll for key, to avoid excessive redraw */
+      if (grub_timeout >= 0) {
+        /* wait for key or timeout change */
+	while (checkkey () < 0 && (time1 = getrtsecs()) == time2 && time1 != 0xFF)
+	  ;
+	/* fall thru on keypress or clock tick */
+      }
 
       /* Check for a keypress, however if TIMEOUT has been expired
 	 (GRUB_TIMEOUT == -1) relax in GETKEY even if no key has been
